### PR TITLE
Add pre release test for packages

### PR DESCRIPTION
**Pull request by maoznir**

### CODE REVIEW

File: test/spec/pkg-spec.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+const getPath = (pkgName, fileName) => path.join(__dirname, '..', '..', 'pkg', pkgName, fileName);
+
+
+/**
+ * Test that a given fileName exist in given pkgName
+ * @param pkgName
+ * @param fileName
+ */
+testFileExists = (pkgName, fileName) => {
```
Comment (const-cloudinary) [inline test/spec/pkg-spec.js]: More appropriate name seems to be assertPathExists (in any case you check both files and dirs)

File: test/spec/pkg-spec.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    expect(result).toEqual(true);
+  })
+};
+
+/**
+ * Test that all given files exist in given pkgName
+ * @param pkgName
+ * @param files
+ */
+testPkg = (pkgName, files) => {
```
Comment (const-cloudinary) [inline test/spec/pkg-spec.js]: I would give more precise name, like verifyPackageConsistency

File: package.json [hunk 3]
```diff
@@ -13,12 +13,13 @@
     "build": "node_modules/.bin/parallel-webpack",
     "doc": "node_modules/.bin/jsdoc -c jsdoc-conf.json",
     "test": "node_modules/.bin/karma start --single-run --browsers=ChromeHeadless",
-    "test:all": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.pkg=${pkg}; done",
-    "test:all:prod": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.minified --cloudinary.pkg=${pkg}; done",
+    "test:all:unminified": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.pkg=${pkg}; done",
```
Comment (asisayag2) [inline package.json]: @maoznir 
Would change to:
test:pkgs - run tests on all packages
test:pkgs:minified - run on minified packages
test:pkgs:verify - verify packages consistency

File: package.json [hunk 4]
```diff
@@ -13,12 +13,13 @@
...
+    "test:all:minified": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.minified --cloudinary.pkg=${pkg}; done",
     "test:types": "node_modules/.bin/tsc",
+    "test:pkg": "jasmine test/spec/pkg-spec.js",
     "pretest": "npm run build",
-    "pretest:all": "npm run build",
-    "pretest:all:prod": "npm run build",
+    "pretest:all:unminified": "npm run build",
```
Comment (asisayag2) [inline package.json]: @maoznir , why do we need those?

File: test/spec/pkg-spec.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  describe(`${pkgName}:`, function () {
+    files.forEach(fileName => testFileExists(pkgName, fileName));
```
Comment (asisayag2) [inline test/spec/pkg-spec.js]: Would build the 'actual' array from the fs and compare to the 'expected' from _files_

File: test/spec/pkg-spec.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
+/**
+ * Test existance of all needed files in each package.
+ * This test file should be run before each release.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const SpecReporter = require('jasmine-spec-reporter').SpecReporter;
+
+// We use 'jasmine-spec-reporter' to get nice logs to console.
```
Comment (const-cloudinary) [inline test/spec/pkg-spec.js]: `We use` ... make it more formal, or remove redundant comment

File: test/spec/pkg-spec.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
+/**
+ * Test existance of all needed files in each package.
```
Comment (const-cloudinary) [inline test/spec/pkg-spec.js]: I would replace Test with Verify

File: test/spec/pkg-spec.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+ * This test file should be run before each release.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const SpecReporter = require('jasmine-spec-reporter').SpecReporter;
+
+// We use 'jasmine-spec-reporter' to get nice logs to console.
+jasmine.getEnv().clearReporters();               // remove default reporter logs
```
Comment (const-cloudinary) [inline test/spec/pkg-spec.js]: redundant comments that just repeat code

File: test/spec/pkg-spec.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+];
+
+/**
+ * Get path of given fileName in given pkgName
+ * @param pkgName
+ * @param fileName
+ * @returns {*}
+ */
+const getPath = (pkgName, fileName) => {
+  if (fileName) {
```
Comment (const-cloudinary) [inline test/spec/pkg-spec.js]: Why don't just
```
if (!fileName) { 
    fileName = '';
}

return path.join(__dirname, '..', '..', 'pkg', pkgName, fileName);
```

File: package.json [hunk 10]
```diff
@@ -13,12 +13,13 @@
...
-    "test:all": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.pkg=${pkg}; done",
-    "test:all:prod": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.minified --cloudinary.pkg=${pkg}; done",
+    "test:pkgs": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.pkg=${pkg}; done",
+    "test:pkgs:minified": "for pkg in core jquery jquery-file-upload; do node_modules/.bin/karma start --single-run --cloudinary.minified --cloudinary.pkg=${pkg}; done",
     "test:types": "node_modules/.bin/tsc",
+    "test:pkgs:verify": "jasmine test/spec/pkg-spec.js",
     "pretest": "npm run build",
-    "pretest:all": "npm run build",
-    "pretest:all:prod": "npm run build",
+    "pretest:pkgs": "npm run build",
```
Comment (asisayag2) [inline package.json]: @maoznir , again, why do we need those?

Comment (maoznir) [reply package.json]: These make sure a build is done before the test.
npm runs them automatically before scripts with the same name without the "pre" prefix.
It's not a must but a convenience. anyway they were not touched in this pr except for matching the name to each related script.

**Pull request merged**